### PR TITLE
Added hidden flag functionality to activities

### PR DESCRIPTION
**Pull request by Muncherkin**
Made it possible to hide activities using a `hidden` flag. Also added functionality to query all activities including or excluding hidden ones.

### CODE REVIEW

**Comment by Muncherkin** [comment]
JAG ÄR JESUS! DET ÄR LÖST! MERGEA MIG!

File: src/api/activity.api.ts [hunk 1]
```diff
@@ -38,7 +39,12 @@ export class ActivityAPI {
           },
           { AND: [{ startDate: { gte: from } }, { endDate: null }] },
         ],
-        AND: { utskott: { in: utskott } },
+        AND: [
+          { utskott: { in: utskott } },
+          {
+            OR: [{ hidden: false }, { hidden: true, ...(hidden ? {} : { hidden: false }) }],
+          },
+        ],
```
Comment (afroborg) [inline src/api/activity.api.ts:47]: This is really hard to follow (and was already before this chaos so not only your fault). Not sure how we could make it better though so maybe just add some clarification because I don't really get how you want it to work?

We have the `hidden` flag, which is `required` in the graphql schema.
- If we send `hidden=true` do we only get the hidden events?
- Same with `hidden=false` for only the non-hidden events?

How would we go about getting all events if it's required in the schema?

File: src/schemas/activity.graphql [hunk 2]
```diff
@@ -43,6 +44,7 @@ input NewActivity{
     utskott: Utskott!
     imageUrl: String
     location: NewLocation
+    hidden: Boolean
```
Comment (afroborg) [inline src/schemas/activity.graphql:47]: why does this need a hidden field?

File: src/schemas/activity.graphql [hunk 3]
```diff
@@ -53,6 +55,7 @@ input ModifiedActivity{
     utskott: Utskott
     imageUrl: String
     location: NewLocation
+    hidden: Boolean
```
Comment (afroborg) [inline src/schemas/activity.graphql:58]: why does this need a hidden field?

**Review by afroborg — CHANGES_REQUESTED** [review]

File: src/api/activity.api.ts (hunk 1 above)
Comment (Muncherkin) [reply src/api/activity.api.ts:47]: I agree that it's completely unreadable, will change! The functionality is opposite to `current` on history queries:

hidden=false, only non hidden activities
hidden=true, all activities.

I'll change the name to `includeHidden`

File: src/schemas/activity.graphql (hunk 2 above)
Comment (Muncherkin) [reply src/schemas/activity.graphql:47]: Do create an activity as hidden or not

File: src/schemas/activity.graphql (hunk 3 above)
Comment (Muncherkin) [reply src/schemas/activity.graphql:58]: Do toggle an activity as hidden or not

File: src/api/activity.api.ts (hunk 1 above)
Comment (Muncherkin) [reply src/api/activity.api.ts]: Solved in latest commit

**Comment by Muncherkin** [comment]
Reran the workflow but the Test annotations persisted even though the tests did actually work after the rerun

File: src/schemas/activity.graphql [hunk 4]
```diff
@@ -4,7 +4,7 @@ scalar DateTime
 
 type Query{
     activity(id: String!) : Activity! 
-    activities(from: DateTime!, to: DateTime!, utskott: [Utskott!]!) : [Activity!]! 
+    activities(from: DateTime!, to: DateTime!, utskott: [Utskott!]!, includeHidden: Boolean!) : [Activity!]!
```
[the remark this answers is not in the document]
Comment (Muncherkin) [reply src/schemas/activity.graphql]: Did this btw!
